mev-boost-rs/src/relay_mux.rs

```diff
@@ -112,11 +112,16 @@ impl BlindedBlockProvider for RelayMux {
             .enumerate()
             .filter_map(|(relay_index, response)| match response {
                 Ok(mut bid) => {
-                    if let Err(err) = validate_bid(&mut bid, &self.context) {
-                        tracing::warn!("invalid signed builder bid: {err} for bid: {bid:?}");
+                    if &self.relays[relay_index].public_key != bid.public_key() {
+                        tracing::warn!("invalid public key for bid: {bid:?}");
                         None
                     } else {
-                        Some((bid, relay_index))
+                        if let Err(err) = validate_bid(&mut bid, &self.context) {
```

Comment:
better to encapsulate this inside `validate_bid`

just pass the `&self.relays[relay_index]` as another parameter to `validate_bid` and check there